macaw-x86-cli/src/Data/Macaw/X86/Symbolic/CLI.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ImplicitParams #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Data.Macaw.X86.Symbolic.CLI
+  ( withX86Hooks
+  ) where
+
+
+import Data.Proxy (Proxy(..))
+
+import Data.Parameterized.NatRepr (knownNat)
+
+import What4.Expr (ExprBuilder)
+
+import Lang.Crucible.Backend qualified as C
+import Lang.Crucible.CLI (SimulateProgramHooks, defaultSimulateProgramHooks)
+import Lang.Crucible.FunctionHandle (newHandleAllocator)
+import Lang.Crucible.LLVM.DataLayout (EndianForm(LittleEndian))
+import Lang.Crucible.LLVM.MemModel qualified as Mem
+import Lang.Crucible.Simulator.ExecutionTree (ExtensionImpl)
+import Lang.Crucible.Syntax.Concrete (ParserHooks)
+
+import Data.Macaw.CFG qualified as DMC
+import Data.Macaw.Memory qualified as DMM
+import Data.Macaw.Symbolic qualified as DMS
+import Data.Macaw.Symbolic.Memory qualified as DMS
+import Data.Macaw.Symbolic.Syntax (machineCodeParserHooks)
+import Data.Macaw.X86 (X86_64)
+import Data.Macaw.X86.Symbolic (newSymFuns, x86_64MacawEvalFn)
+import Data.Macaw.X86.Symbolic.Syntax (x86ParserHooks)
+
+ -- | Small helper for providing LLVM language-specific hooks, e.g., for use with
+-- 'Lang.Crucible.CLI.execCommand'.
+withX86Hooks ::
+  ((?parserHooks :: ParserHooks (DMS.MacawExt X86_64)) =>
+   (forall sym bak t st fs. 
+     ( C.IsSymBackend sym bak
+     , sym ~ ExprBuilder t st fs
+     ) =>
+     bak ->
+     IO (ExtensionImpl () sym (DMS.MacawExt X86_64))) ->
+    SimulateProgramHooks (DMS.MacawExt X86_64) ->
+    IO a) ->
+  IO a
+withX86Hooks k = do
+  ha <- newHandleAllocator
+  mvar <- Mem.mkMemVar "llvm_memory" ha
```

Comment:
Other places in `macaw` prefix the label with the package that created the LLVM memory, e.g.,

https://github.com/GaloisInc/macaw/blob/5453edeaa791b32bcf8a67b9584a0ec06b07c6b3/refinement/src/Data/Macaw/Refinement/SymbolicExecution.hs#L494
https://github.com/GaloisInc/macaw/blob/5453edeaa791b32bcf8a67b9584a0ec06b07c6b3/symbolic/src/Data/Macaw/Symbolic/Testing.hs#L542

Perhaps it would be worth following suit and labelling this `macaw-x86-cli:llvm_memory` or some such.